src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -421,7 +448,73 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     }
   }
 
-  override def handleCallRestarts(restartableWorkflows: Seq[RestartableWorkflow])(implicit ec: ExecutionContext): Future[Any] = Future("FIXME")
+  /**
+   * <ul>
+   *   <li>Any execution in Failed should fail the restart.</li>
+   *   <li>Any execution in Aborted should fail the restart.</li>
+   *   <li>Scatters in Starting should fail the restart.</li>
+   *   <li>Collectors in Running should be set back to NotStarted.</li>
+   *   <li>Calls in Starting should be rolled back to NotStarted.</li>
+   *   <li>Calls in Running with no job key should be rolled back to NotStarted.</li>
+   * </ul>
+   *
+   * Calls in Running *with* a job key should be left in Running.  The WorkflowActor is responsible for
+   * resuming the call actors for these calls.
+   */
+  override def prepareForRestart(restartableWorkflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    lazy val tag = s"Workflow ${restartableWorkflow.id.shortString}:"
+
+    def handleExecutionStatuses(executions: Traversable[Execution]): Future[Unit] = {
+
+      val executionsByStatus = executions.groupBy(_.status) map { case (k, v) => ExecutionStatus.withName(k) -> v }
+      def stringifyExecutions(executions: Traversable[Execution]): String = {
+        executions.toSeq.sortWith((lt, rt) => lt.callFqn < rt.callFqn || (lt.callFqn == rt.callFqn && lt.index < rt.index)).mkString(" ")
+      }
+
+      if (executionsByStatus.contains(ExecutionStatus.Failed)) {
+        val failedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Failed).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Failed status: " + failedExecutions))
+      }
+      else if (executionsByStatus.contains(ExecutionStatus.Aborted)) {
+        val abortedExecutions = stringifyExecutions(executionsByStatus.get(ExecutionStatus.Aborted).get)
+        Future.failed(new Throwable(s"$tag Cannot restart, found executions in Aborted status: " + abortedExecutions))
+      }
+      else {
+        // Cromwell currently does not persist the types of executions.  Scatters are identified by this magic
+        // "$scatter_" string.
+        val (scatters, nonScatters) = executions partition { _.callFqn.contains("$scatter_") }
+        if (scatters.exists(_.status.toExecutionStatus == ExecutionStatus.Starting)) {
+          val startingScatters = stringifyExecutions(scatters)
+          Future.failed(new Throwable(s"$tag Cannot restart, found scatters in Starting status: " + startingScatters))
```

Comment:
Here are the comments I left in WorkflowActor.  Basically it's not safe to do without adding some more queries that will see if the scattered calls have already been created, and that's probably not worth doing at this point:

    // Assumptions for restart:
    // The only states for a scatter are Starting and Done, and if it's Starting then it's not Done and
    // we don't know without some additional db queries whether scattered calls have been created.
    //
    // For now we'll fail the workflow if we find a scatter in Starting and force manual fixup:
    // either set the scatter back to NotStarted or mark it as Done as appropriate.  This should be an
    // unlikely scenario for Cromwell to find itself in as this doesn't require any interaction with JES and
    // should execute very quickly.  It is definitely possible to either bracket the work below in a transaction
    // or add the queries to diagnose this more specifically and do fixups in an automated way, but that's
    // probably not time well spent at this point.